packages/utils/src/ColorUtils.ts

```diff
@@ -32,5 +48,93 @@ class ColorUtils {
       (color[0] * 299 + color[1] * 587 + color[2] * 114) / 1000
     );
   }
+
+  /**
+   * Normalize a css color to 8 character hex value. If the color can't be resolved,
+   * return the original color string.
+   * @param colorString The color string to normalize
+   */
+  static normalizeCssColor(colorString: string): string {
+    const maybeRgbOrRgba = ColorUtils.asRgbOrRgbaString(colorString);
+    if (maybeRgbOrRgba == null) {
+      return colorString;
+    }
+
+    const rgba = ColorUtils.parseRgba(maybeRgbOrRgba);
+    if (rgba === null) {
+      return colorString;
+    }
+
+    return ColorUtils.rgbaToHex8(rgba);
+  }
+
+  /**
+   * Parse a given `rgb` or `rgba` css expression into its constituent r, g, b, a
+   * values. If the expression cannot be parsed, it will return null.
+   * Note that this parser is more permissive than the CSS spec and shouldn't be
+   * relied on as a full validation mechanism. For the most part, it assumes that
+   * the input is already a valid rgb or rgba expression.
+   *
+   * e.g. `rgb(255, 255, 255)` -> `{ r: 255, g: 255, b: 255, a: 1 }`
+   * e.g. `rgba(255, 255, 255, 0.5)` -> `{ r: 255, g: 255, b: 255, a: 0.5 }`
+   * @param rgbOrRgbaString The rgb or rgba string to parse
+   */
+  static parseRgba(
+    rgbOrRgbaString: string
+  ): { r: number; g: number; b: number; a: number } | null {
+    const [, name, args] = /^(rgba?)\((.*?)\)$/.exec(rgbOrRgbaString) ?? [];
+    if (name == null) {
+      return null;
+    }
+
+    // Split on spaces, commas, and slashes. Note that this more permissive than
+    // the CSS spec in that slashes should only be used to delimit the alpha value
+    // (e.g. r g b / a), but this would match r/g/b/a. It also would match a mixed
+    // delimiter case (e.g. r,g b,a). This seems like a reasonable tradeoff for the
+    // complexity that would be added to enforce the full spec.
+    const tokens = args.split(/[ ,/]/).filter(Boolean);
+
+    if (tokens.length < 3) {
+      return null;
+    }
+
+    const [r, g, b, a = 1] = tokens.map(Number);
+
+    return {
+      r,
+      g,
+      b,
+      a,
+    };
```

Comment:
I'd much prefer without the custom line breaks here
```suggestion
    return { r, g, b, a };
```

Reply:
ah yeah. I'll fix on next PR